### PR TITLE
US2_2T1 Implement API for admin to get all and get one transaction

### PR DESCRIPTION
**Pull request by ShamaW**
- Add Transaction controllers 
- Add function to get one transaction and get all the transactions

### CODE REVIEW

**Review by Yiwahpsp — COMMENTED** [review]
Have 2 comments please take a look, other looks great

File: controllers/transactions.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  }
+};
+
+// @desc:    Get a single transaction with an id
+// @route:   GET /api/v1/transactions/:id
+// @access:  Private
+exports.getTransaction = async (req, res, next) => {
+  try {
+    const transaction = await Transaction.findById(req.params.id);
+
```
[the remark this answers is not in the document]
Comment (FaiiFaiipuipui) [reply controllers/transactions.js:100]: resolved yet?

File: controllers/transactions.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      path: "appointment",
+      select: "user campground",
+    });
+  } else {
+    // If you are an admin, you can see all!
+    if (req.params.transactionId) {
+      console.log(req.params.transactionId);
+      query = Transaction.find({
+        campground: req.params.transactionId,
+      })
```
[the remark this answers is not in the document]
Comment (FaiiFaiipuipui) [reply controllers/transactions.js:55]: resolved yet?

File: controllers/transactions.js
Comment (FaiiFaiipuipui) [inline controllers/transactions.js:1]: are there any way for admin/user to get their transaction slip?

File: controllers/transactions.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  // Pagination
+  const page = parseInt(req.query.page, 10) || 1;
+  const limit = parseInt(req.query.limit, 10) || 10;
+  const startIndex = (page - 1) * limit;
+  const endIndex = page * limit;
+  const total = await Transaction.countDocuments();
+
+  query = query.skip(startIndex).limit(limit);
+
+  if (req.user.role !== "admin") {
```
Comment (FaiiFaiipuipui) [inline controllers/transactions.js]: from routes/transactions.js only the admins are allowed to access this API. are user suppose to be able to access this API too?

**Review by FaiiFaiipuipui — CHANGES_REQUESTED** [review]

**Comment by ShamaW** [comment]
Try to make some change according to you guy's suggestion. How does it look?

**Review by Yiwahpsp — APPROVED** [review]
look great!!

**Review by BoondaPanich — APPROVED** [review]
lgtm

**Review by BoondaPanich — APPROVED** [review]
Looks good kub 👍👍

File: routes/campgrounds.js [hunk 4]
```diff
@@ -16,6 +17,7 @@ const router = express.Router();
 const { protect, authorize } = require("../middleware/auth");
 
 router.use('/:campgroundId/appointments/', appointmentRouter);
+router.use('/:campgroundId/transactions/', transactionRouter);
```
Comment (FaiiFaiipuipui) [inline routes/campgrounds.js]: Is this route suppose to exist?

**Review by FaiiFaiipuipui — APPROVED** [review]
Please review comment, other than that I think it's all good😺👍

**Review by Yiwahpsp — APPROVED** [review]
look goodddd

**Pull request merged**